src/rgw/rgw_auth_s3.h

```diff
@@ -578,11 +647,21 @@ static inline const char* get_v4_exp_payload_hash(const req_info& info)
   return expected_request_payload_hash;
 }
 
-static inline bool is_v4_payload_unsigned(const char* const exp_payload_hash)
+static inline bool is_traditional_v4_unsigned_payload(const char* const exp_payload_hash)
 {
   return boost::equals(exp_payload_hash, AWS4_UNSIGNED_PAYLOAD_HASH);
 }
 
+static inline bool is_v4_payload_unsigned(const char* const exp_payload_hash)
+{
+  return boost::contains(exp_payload_hash, "UNSIGNED-PAYLOAD");
+}
+
+static inline bool have_checksum_trailer(const char* const exp_payload_hash)
+{
+  return boost::algorithm::ends_with(exp_payload_hash, "TRAILER");
```

Comment:
ok?  I don't to be honest find the mixture of behaviors in v4 signing and checksums all that orthognal, but I was just hacking at the naming here.  initially, I did think the trailer was *for* checksum--and basically, the term trailer clearly was introduced with checksum, for what it's worth.  since it also can contain a chunk-signature, it does seem like it's "all trailer stuff"